src/app/Components/ArtworkGrids/ArtworkGridItem.tsx

```diff
@@ -208,18 +208,30 @@ export const Artwork: React.FC<ArtworkProps> = ({
     if (trackTap) {
       trackTap(artwork.slug, itemIndex)
     } else if (contextScreenOwnerType) {
-      const genericTapEvent = tappedMainArtworkGrid({
-        contextScreen,
-        contextScreenOwnerType: contextScreenOwnerType,
-        contextScreenOwnerId,
-        contextScreenOwnerSlug,
-        destinationScreenOwnerId: artwork.internalID,
-        destinationScreenOwnerSlug: artwork.slug,
+      const genericTapEvent: TappedMainArtworkGrid = {
+        action: ActionType.tappedMainArtworkGrid,
+        context_module: ContextModule.artworkGrid,
+        context_screen: contextScreen,
+        context_screen_owner_type: contextScreenOwnerType,
+        context_screen_owner_id: contextScreenOwnerId,
+        context_screen_owner_slug: contextScreenOwnerSlug,
+        destination_screen_owner_type: OwnerType.artwork,
+        destination_screen_owner_id: artwork.internalID,
+        destination_screen_owner_slug: artwork.slug,
         position: itemIndex,
-        // This is always a string; types are incorrect
-        sort: String(filterParams?.sort),
         query: contextScreenQuery,
-      })
+        sort: filterParams?.sort,
+        type: "thumbnail",
+      }
+
+      if (
+        AREnablePartnerOfferSignals &&
+        collectorSignals?.partnerOffer?.isAvailable &&
+        !artwork.sale?.isAuction
+      ) {
+        genericTapEvent.signal_label = "Limited-Time Offer"
+      }
```

Comment:
As of today partner offer + auction artwork are mutually exclusive so this check should not be necessary.

Reply:
```suggestion
      if (
        AREnablePartnerOfferSignals &&
        !!collectorSignals?.partnerOffer
      ) {
        genericTapEvent.signal_label = "Limited-Time Offer"
      }
```
Technically this should be all we need - `isAvailable` refers to the artwork being available for purchase, and the partner offer is only returned if the artwork is purchaseable and the partner offer is still active.